plugins/modules/azure_rm_aks.py

```diff
@@ -676,6 +720,7 @@ def create_addon_profiles_spec():
     vnet_subnet_id=dict(type='str'),
     availability_zones=dict(type='list', elements='int', choices=[1, 2, 3]),
     os_type=dict(type='str', choices=['Linux', 'Windows']),
+    os_sku=dict(type='str', choices=['Ubuntu', 'Windows2019', 'Windows2022']),
```

Comment:
The SDK only supports two choices for this parameters ("Ubuntu", "CBLMariner")

Reply:
Are you sure that only 2 params are supported? It did work for me... to create windows nodes.

https://github.com/Azure/azure-sdk-for-python/blob/81fa2f833163808415256a0f6722d0a023cf50eb/sdk/containerservice/azure-mgmt-containerservice/azure/mgmt/containerservice/v2023_04_01/models/_container_service_client_enums.py#L294

https://github.com/Azure/azure-sdk-for-python/blob/81fa2f833163808415256a0f6722d0a023cf50eb/sdk/containerservice/azure-mgmt-containerservice/azure/mgmt/containerservice/v2021_07_01/models/_models_py3.py#L2509

It looks like any string is valid too! Right?

Do you think it is correct to alter the list to all types from https://github.com/Azure/azure-sdk-for-python/blob/81fa2f833163808415256a0f6722d0a023cf50eb/sdk/containerservice/azure-mgmt-containerservice/azure/mgmt/containerservice/v2023_04_01/models/_container_service_client_enums.py#L294